File: src/main/java/me/kuku/yuq/logic/QQAILogic.java
package me.kuku.yuq.logic;

import com.IceCreamQAQ.Yu.annotation.AutoBind;
import me.kuku.yuq.pojo.Result;

import java.io.IOException;

@AutoBind
public interface QQAILogic {
    boolean pornIdentification(String imageUrl) throws IOException;
    String generalOCR(String imageUrl) throws IOException;
    String textChat(String question, String session) throws IOException;
    Result<byte[]> voiceSynthesis(String text) throws IOException;
}

File: src/main/kotlin/me/kuku/yuq/logic/QQAILogic.kt
package me.kuku.yuq.logic

import com.IceCreamQAQ.Yu.annotation.AutoBind
import me.kuku.yuq.pojo.CommonResult

@AutoBind
interface QQAILogic {
    fun pornIdentification(imageUrl: String): Boolean
    fun generalOCR(imageUrl: String): String
    fun textChat(question: String, session: String): String
    fun echoSpeechRecognition(url: String): String
    fun aiLabSpeechRecognition(url: String): String
    fun voiceSynthesis(text: String): CommonResult<ByteArray>
}